gatsby-node.js

```diff
@@ -252,40 +252,99 @@ exports.onCreateWebpackConfig = ({ actions, getConfig }, { ...options }) => {
+                        image {
+                            id
+                            imageFile {
+                                childImageSharp {
+                                    gatsbyImageData
+                                }
+                            }
+                        }
+                    }
+                    main_image {
+                        id
+                        imageFile {
+                            childImageSharp {
+                                gatsbyImageData
+                            }
+                        }
+                    }
+                    tags {
+                        id
+                        tags_id {
+                            id
+                            tag_name
+                        }
+                    }
+                    footer_banners {
+                        id
+                        cta_url
+                        name
+                        desktop_banner_image {
+                            id
+                            imageFile {
+                                childImageSharp {
+                                    gatsbyImageData
+                                }
+                            }
+                        }
+                        mobile_banner_image {
+                            id
+                            imageFile {
+                                childImageSharp {
+                                    gatsbyImageData
+                                }
+                            }
+                        }
+                    }
+                    side_banners {
+                        id
+                        cta_url
+                        name
+                        banner_image {
+                            id
+                            imageFile {
+                                childImageSharp {
+                                    gatsbyImageData
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    `)
 
-//     articles.forEach((article) => {
-//         createPage({
-//             path: `/blog/articles/${article.article_url}`,
-//             component: articleTemplate,
-//             context: {
-//                 locale: 'en',
-//                 pathname: `/blog/articles/${article.article_url}`,
-//                 slug: article.article_url,
-//             },
-//         })
-//     })
-// }
+    if (result.errors) {
+        reporter.panic(result.errors)
+    }
+    const articles = result.data.directus.blog
+    articles.forEach((article) => {
+        createPage({
+            path: `/academy/blog/articles/${article.id}`,
+            component: articleTemplate,
+            context: {
+                locale: 'en',
+                pathname: `/academy/blog/articles/${article.id}`,
+                slug: article.id,
+            },
+        })
+    })
+}
```

Comment:
```suggestion
    const blog = result.data.directus.blog
    blog.forEach((blog_post) => {
        createPage({
            path: `/academy/blog/${blog_post.slug}`,
            component: articleTemplate,
            context: {
                locale: 'en',
                pathname: `/academy/blog/${blog_post.slug}`,
                slug: blog_post.slug,
            },
        })
    })
}
```

Reply:
You will also need the slug in the query.

Reply:
Amended in latest commit